tests/test_apypie.py

```diff
@@ -80,6 +80,26 @@ def test_init_with_existing_cachedir(fixture_dir, requests_mock, tmpdir):
     assert tmpdir.join('apypie/https___api.example.com/v1/default.json').check(exists=0)
 
 
+def test_init_with_bad_cache(fixture_dir, requests_mock, tmpdir):
+    with fixture_dir.join('dummy.json').open() as read_file:
+        data = json.load(read_file)
+    requests_mock.get('https://api.example.com/apidoc/v1.json', json=data)
+
+    old_environ = os.environ.copy()
+    os.environ['XDG_CACHE_HOME'] = tmpdir.strpath
+
+    json_path = tmpdir.join('apypie/https___api.example.com/v1/deadc0ffee.json')
```

Comment:
Isn't this cleaner?
```suggestion
    json_path = tmpdir.join('apypie', 'https___api.example.com', 'v1', 'deadc0ffee.json')
```

Reply:
given this is also the case in other tests, I'll merge now and refactor all in one go.